printf.c

```diff
@@ -111,6 +111,45 @@
 #define FLAGS_PRECISION (1U << 10U)
 #define FLAGS_ADAPT_EXP (1U << 11U)
 
+// math constants
+#define BASE_2U    2U
+#define BASE_8U    8U
+#define BASE_10U  10U
+#define BASE_16U  16U
+#define SHIFT_52U 52U
+#define FLOATING_HALF 0.5

```

Comment:
This would work better as a typed constant, i.e. a `static constexpr`, so specifying the type in the name would not be necessary.

Reply:
In my next pull request (coming soon), I've changed all these to const and constexpr as appropriate.

Reply:
My next update will include new test cases and bug fixes in floating point, as well as addressing all issues mentioned in your comments here.